pyccel/ast/datatypes.py

```diff
@@ -8,7 +8,7 @@
 from sympy import Eq, Ne, Lt, Gt, Le, Ge
 
 
-default_precision = {'real': 8, 'int': 8, 'complex': 8, 'bool':1, 'float':8}
+default_precision = {'real': 8, 'int': 8, 'complex': 8, 'bool':4, 'float':8}
```

Comment:
Interesting that we should use 4 Bytes to store a boolean variable. Can you explain why we need this?

Reply:
If you have a function with an argument of type logical(kind=1) and you pass .True. then gfortran will complain that the types don't match

Reply:
What worries me is that we have no way to determine the kind parameter value in Fortran. For integers there is `selected_int_kind`, for floats `selected_real_kind` and for strings `selected_char_kind`. But for booleans there is no equivalent function.

Reply:
If we want to ensure compatibility between integer and logical variables, maybe these should have the same kind. But I say "maybe" here! I suspect that we should always print logicals without specifying their kind, hence using the default kind provided by the machine. I wonder how f2py handles this issue.

Reply:
@chris @EmilyBourne Thanks for all this work!
You need to fix some small Codacy issue before we can proceed with this merge.
And I am also waiting for @saidctb approval for this.

Reply:
I think C handles it the same way. There is no smaller unit than 4 Bytes on 32bit x86 and they kept it for x64 also probably for compatibility. So internally you always reserve an integer when you want a bool.